### PR TITLE
Improve with_hierarchy()

### LINKED ISSUE #14: Invalid call to DataBank?
Line 76 of `with_hierarchy.R` has this code:
```
    # Handle the case where a t-table is provided directly
    # (it has to be reshaped!)
    hierarchy_object <-
      DataBank::DB_hierarchy(
        using = using,
        verbose = verbose)
```

I encountered it because the package does not Import or Depend on DataBank in the DESCRIPTION file, which is causing errors during R CMD CHECK.

The issue actually turns out to be more significant than a naming issue. It appears that DataBank does not have a function called `DB_hierarchy` at all!
```
> library(DataBank)
Loading required package: welcome
Today is Thu Apr  8 11:50:47 2021
Loading required package: ensurer
Loading required package: memoise
Loading required package: Ingres
Loading required package: cacher
Error: package ‘tbltools’ required by ‘Ingres’ could not be found
> DataBank::DB_hierarchy
Error: 'DB_hierarchy' is not an exported object from 'namespace:DataBank'
> DataBank:::DB_hierarchy
Error in get(name, envir = asNamespace(pkg), inherits = FALSE) : 
  object 'DB_hierarchy' not found
```

@dholstius can you please comment? I notice that DataBank exports two functions with similar names: `DB_category_hierarchy` and `DB_SIC_hierarchy`. Is it possible that `DB_hierarchy` used to exist but was deleted in favor of these other functions? What should this line be?

### PR DESCRIPTION
**Pull request by dholstius**
- Fixes #14 
- Adds `DataBank` to `Suggests`. There may still be some issues with `DataBank` functions not being properly namespaced, and it seems likely that it's better off in `Imports` than `Suggests`. Leaving that for a subsequent issue/PR.

### CODE REVIEW

**Comment by arilamstein** [comment]
@dholstius FYI It looks like this PR now has conflicts with master. They seem minor and I will work to resovle them before approving.

File: DESCRIPTION [hunk 1]
```diff
@@ -51,7 +51,6 @@ RoxygenNote: 7.1.1
 Roxygen: list(markdown = TRUE)
 Suggests: 
     testthat (>= 2.1.0),
-    data.table,
```
Comment (arilamstein) [inline DESCRIPTION]: data.table is already included in Imports, so it does not need to be in suggests.

**Review by arilamstein — COMMENTED** [review]
I think that your solution works. But I think that there might be a more elegant solution. It appears that the only usage of the DataBank package in this package is now line 79 of with_hierarchy.R. This appears immediately after a call to .Defunct().

The way .Defunct() works is that it triggers an error. This means that the actual calls to the DataBank package below will never get executed. If we comment out those lines (or simply delete them), then we can completely remove DataBank from the DESCRIPTION file. As long as there are actual calls to the DataBank package in this package, we need to include DataBank in the package's DESCRIPTION file somehow (even if the code is never called).

I mention this option because, as a rule of thumb, it is best to keep a package's dependencies to a minimum. And if this package is, in effect, no longer dependent on the DataBank package, it might be best to simply remove the dependency altogether.

What do you think?

File: DESCRIPTION (hunk 1 above)
Comment (arilamstein) [reply DESCRIPTION]: To clarify: having a package listed in two sections leads to an error/warning in R CMD CHECK.

File: tests/testthat/test-with_hierarchy.R [hunk 2]
```diff
@@ -30,3 +30,16 @@ test_that("with_hierarchy (using BY2011 and test_hierarchy)", {
       test_hierarchy)
 
 })
+
+test_that("using=t0327", {
+
+  test_data <-
+    tibble(cat_id = 838)
+
+  expect_warning(
```
Comment (arilamstein) [inline tests/testthat/test-with_hierarchy.R:39]: I believe that `.Defunct()` creates an error, not a warning. So this should probably be `expect_error`. For example:
```
test_defunct = function()
{
  .Defunct()
  
  print("HI!")
}

test_defunct()
```
Results in:
```
Error: 'test_defunct' is defunct.
See help("Defunct")
Called from: .Defunct()
```

**Comment by dholstius** [comment]
I do want to .Deprecate() for a bit before employing .Defunct(). Did I do the latter instead of the former? Sorry, on my phone

**Comment by arilamstein** [comment]
> I do want to .Deprecate() for a bit before employing .Defunct(). Did I do the latter instead of the former? Sorry, on my phone

You are right! I'm not sure how / why I misread that, but I did. You are using `.Deprecated` and that does indeed trigger a warning, and not an error.

**Comment by arilamstein** [comment]
I think that other issue I mentioned (about `data.table` being included twice causing a problem) was valid. But I submitted a change that fixed that. Since the only other issue I mentioned was not in fact an issue, I went ahead with the merge.

**Pull request merged**